src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/RenderInputParams/InputComponent/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import Col from '../../../../../../../../../components/layout/Col'
+import Field from '../../../../../../../../../components/forms/Field'
+import Checkbox from '../../../../../../../../../components/forms/Checkbox'
+import TextField from '../../../../../../../../../components/forms/TextField'
+import {
+  composeValidators,
+  mustBeEthereumAddress,
+  required,
+} from '../../../../../../../../../components/forms/validator'
+
+type Props = {
+  type: string
+  keyValue: string
+  placeholder: string
+}
+
+const InputComponent = ({ type, keyValue, placeholder }: Props) => {
+  if (!type) {
+    return null
+  }
+  switch (type) {
+    case 'bool': {
+      const inputProps = {
```

Comment:
this could've been defined outside `switch/case` statement

Reply:
btw feels unusual to see switch/case inside a component, looks like you're missing writing reducers 😄